src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Helpers/BackupScheduleHelpers.ps1

```diff
@@ -0,0 +1,346 @@
+		# Validate TimeZone
+
+		if(-not $TimeZone) {
+			if ($manifest.allowedSubProtectionPolicyTypes.Count -gt 1) {
+				# SAPHANA case
+				if(-not $Policy.Setting.TimeZone) {
+					$errormsg = "TimeZone cannot be empty for FullBackup"
+					throw $errormsg
+				}
+				else {
+					$TimeZone = $Policy.Setting.TimeZone
+				}
+			}
+			else {
+				# AzureVM case
+			}
+		}
+
+		# Validate PolicySubType
+		
+		if ($PolicySubType) {
+			if (-not $manifest.supportsEnhanced) {
+				$errormsg = "PolicySubType is not supported for DatasourceType " + $DatasourceType
+				throw $errormsg
+			}
+		}
+
+		if ($manifest.allowedSubProtectionPolicyTypes.Count -eq 1) {
+			if (-not $PolicySubType) {
+				if (-not $Policy.PolicyType) {
+					$errormsg = "PolicySubType cannot be empty for AzureVM FullBackup"
+				}
+				else {
+					$PolicySubType = $Policy.PolicyType
+				}
+			}
+		}
+		
+		# Validate Hourly specific parameters
+		
+		if ($Interval -or $ScheduleWindowDuration) {
+			$parametersEntered = [System.Collections.ArrayList]::new()
+			if ($Interval) {
+				$parametersEntered.Add("Interval")
+			}
+			if ($ScheduleWindowDuration) {
+				$parametersEntered.Add("ScheduleWindowDuration")
+			}
+
+			if (-not ($BackupFrequency -eq "Hourly" -and $PolicySubType -eq "Enhanced")) {
+				$errormsg = $parametersEntered + " can only be set for Enhanced Hourly backups"
+				throw $errormsg
+			}
+		}
+
+		if ($BackupFrequency -eq "Hourly") {
+			if(-not $Interval) {
+				if(-not $Policy.SchedulePolicy.HourlySchedule.Interval) {
+					$errormsg = "Interval cannot be empty for Hourly backups"
+				}
+				else {
+					$Interval = $policyObject.SchedulePolicy.HourlySchedule.Interval
+				}
+			}
+
+			if(-not $ScheduleWindowDuration) {
+				if(-not $Policy.SchedulePolicy.HourlySchedule.ScheduleWindowDuration) {
+					$errormsg = "ScheduleWindowDuration cannot be empty for Hourly backups"
+				}
+				else {
+					$ScheduleWindowDuration = $policyObject.SchedulePolicy.HourlySchedule.ScheduleWindowDuration
+				}
+			}
+		}
+
+		# Validate Log Backup parameters
+
+		if($LogBackup) {
+			if (-not $manifest.supportsLogBackup) {
+				$errormsg = "LogBackup is not supported for DatasourceType " + $DatasourceType
+				throw $errormsg
+			}
+
+			if (-not $LogBackupFrequency) {
+				$errormsg = "LogBackupFrequency cannot be empty for LogBackup"
+				throw $errormsg
+			}
+		}
+
+		# Validate Differential parameters	
+		
+		if ($EnableDifferentialBackup -or $DifferentialRunDay -or $DifferentialRunTime) {
+			$parametersEntered = [System.Collections.ArrayList]::new()
+			if ($EnableDifferentialBackup) {
+				$parametersEntered.Add("EnableDifferentialBackup")
+			}
+			if ($DifferentialRunDay) {
+				$parametersEntered.Add("DifferentialRunDay")
+			}
+			if ($DifferentialRunTime) {
+				$parametersEntered.Add("DifferentialRunTime")
+			}
+			
+			if (-not ($manifest.supportsDifferentialBackup -and $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunFrequency -eq "Weekly")) {
```

Comment:
should the if condition be something like 

if (-not (**$manifest.allowedSubProtectionPolicyTypes -contains "Differential"** -and $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunFrequency -eq "Weekly")) {